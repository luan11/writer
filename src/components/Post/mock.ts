export const postPropsMock = {
  id: 0,
  content: 'My content',
  createdAt: new Date().toLocaleDateString(),
  likes: {
    count: 0,
    reacted: false
  },
  loves: {
    count: 0,
    reacted: false
  },
  author: 'John Doe'
};