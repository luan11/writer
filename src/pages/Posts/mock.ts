export const postsMock = [
  {
    id: 1,
    content: 'Content 1',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
  {
    id: 2,
    content: 'Content 2',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
  {
    id: 3,
    content: 'Content 3',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
  {
    id: 4,
    content: 'Content 4',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
  {
    id: 5,
    content: 'Content 5',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
  {
    id: 6,
    content: 'Content 6',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
  {
    id: 7,
    content: 'Content 7',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
  {
    id: 8,
    content: 'Content 8',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
  {
    id: 9,
    content: 'Content 9',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
  {
    id: 10,
    content: 'Content 10',
    createdAt: new Date().toISOString(),
    likes: 5,
    activeUserLikedIt: 1,
    loves: 0,
    activeUserLovedIt: 0,
    author: {
      id: 1,
      username: 'John Doe',
    },
  },
];