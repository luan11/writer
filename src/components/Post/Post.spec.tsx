import { render, screen } from '@testing-library/react';

import { Post } from '.';
import { postPropsMock } from './mock';

const props = postPropsMock;

describe('<Post />', () => {
  it('should render Post correctly', () => {
    const regexp = new RegExp(`${props.likes.count}|${props.loves.count}`);

    render(<Post {...props} />);

    expect(screen.getByText(props.createdAt, { selector: 'time' })).toBeInTheDocument();
    expect(screen.getByText(props.content, { selector: 'p' })).toBeInTheDocument();
    expect(screen.getByText(props.author, { selector: 'p' })).toBeInTheDocument();
    expect(screen.getAllByRole('button', { name: regexp })).toHaveLength(2);
  });

  it('should match snapshot', () => {
    const { container } = render(<Post {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});