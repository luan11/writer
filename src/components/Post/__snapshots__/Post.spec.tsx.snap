// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should match snapshot 1`] = `
<article
  class="sc-bdnxRM bVrGuJ"
>
  <time
    class="sc-gtsrHT giwgFo"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
      />
    </svg>
    06/06/2021
  </time>
  <p
    class="sc-dlnjwi kVplKD"
  >
    My content
  </p>
  <p
    class="sc-hKFxyN gDZyIV"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 496 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
      />
    </svg>
    John Doe
  </p>
  <div
    class="sc-eCApnc eXUwJv"
  >
    <button
      class="sc-jSFjdj gWjvDJ"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
        />
      </svg>
      0
    </button>
    <button
      class="sc-jSFjdj gWjvDJ"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M885.9 533.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.4-65.5-111.1a67.67 67.67 0 0 0-34.3-9.3H572.4l6-122.9c1.4-29.7-9.1-57.9-29.5-79.4A106.62 106.62 0 0 0 471 99.9c-52 0-98 35-111.8 85.1l-85.9 311H144c-17.7 0-32 14.3-32 32v364c0 17.7 14.3 32 32 32h601.3c9.2 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7-.2-12.6-2-25.1-5.6-37.1zM184 852V568h81v284h-81zm636.4-353l-21.9 19 13.9 25.4a56.2 56.2 0 0 1 6.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 0 1 6.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 0 1 6.9 27.3c0 22.4-13.2 42.6-33.6 51.8H329V564.8l99.5-360.5a44.1 44.1 0 0 1 42.2-32.3c7.6 0 15.1 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.2 32.1-19.6 43z"
        />
      </svg>
      0
    </button>
  </div>
</article>
`;
